### PR TITLE
[ECMS-6760][Site Explorer] Inconsistence for displaying version numbe…

### PR DESCRIPTION
**Pull request by AtiaZ**
…r between Activity stream, Version Info and Views

Description : 
- When upload Version 0, it's the root version.
- Add the rootVersion as baseVersion.
-Enable the ViewVersion of this baseVersion.

### CODE REVIEW

File: apps/portlet-explorer/src/main/webapp/groovy/webui/component/explorer/versions/UIVersionInfo.gtmpl [hunk 1]
```diff
@@ -38,7 +31,7 @@
 		  }
 		  label.append("</label>") ;
 		  
-		  String show = "Version: " + node.getName() + " " + label.toString() + baseVersion;
+		  String show = "Version: " + node.getName() + " " + label.toString();
```
Comment (trongtt) [inline apps/portlet-explorer/src/main/webapp/groovy/webui/component/explorer/versions/UIVersionInfo.gtmpl]: should support i18n

File: apps/portlet-explorer/src/main/webapp/groovy/webui/component/explorer/versions/UIVersionInfo.gtmpl [hunk 2]
```diff
@@ -83,7 +72,20 @@
 		<table class="uiGrid table no-border-cell table-hover table-striped">
 			<%
 				String clazz = "" ;
-				writeNodes(rootNode.getChildren(), clazz, 0 );
+				VersionNode rootNode = uicomponent.getRootVersionNode();
+				String baseVersion = "" ;
+				baseVersion = " (Base Version)";
+                String show1 = "Version: " + "0" + baseVersion;
```
Comment (trongtt) [inline apps/portlet-explorer/src/main/webapp/groovy/webui/component/explorer/versions/UIVersionInfo.gtmpl]: need i18n support

File: apps/portlet-explorer/src/main/webapp/groovy/webui/component/explorer/versions/UIVersionInfo.gtmpl [hunk 3]
```diff
@@ -83,7 +72,20 @@
...
+                println """
+                  <tr><td>$show1</td>
+                """;
+                println """
+                  <td class="right">
+                """;
+                println "<a class=\"actionIcon\" onclick=\"" + uicomponent.event("AddLabel",   rootNode.getPath()) + "\" rel=\"tooltip\" data-placement=\"bottom\" title=\"" + _ctx.appRes("UIVersionInfo.tooltip.addLabel") + "\"><i class=\"uiIconPlus uiIconLightGray\"></i></a>" ;
+                println "<a class=\"actionIcon\" onclick=\"" + uicomponent.event("ViewVersion",   rootNode.getPath()) + "\" rel=\"tooltip\" data-placement=\"bottom\" title=\"" + _ctx.appRes("UIVersionInfo.tooltip.viewVersion") + "\"><i class=\"uiIconPreview uiIconLightGray\"></i></a>" ;
+                println "</tr>" ;
+				writeNodes(rootNode.getChildren(), clazz, 1);
```
Comment (trongtt) [inline apps/portlet-explorer/src/main/webapp/groovy/webui/component/explorer/versions/UIVersionInfo.gtmpl]: Indentation is not consistent

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/UIDocumentNodeList.java [hunk 4]
```diff
@@ -298,7 +298,8 @@ public String getLabel(String id)  {
    */
   public String getVersionNumber(Node file) throws Exception {
     if (file.isNodeType(NodetypeConstant.MIX_VERSIONABLE)) {
-      return "V" + file.getVersionHistory().getAllVersions().getSize();
+	  int size = (int)file.getVersionHistory().getAllVersions().getSize()-1;
+      return "V" + size;
```
Comment (trongtt) [inline core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/UIDocumentNodeList.java]: Indentation

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/versions/UIVersionInfo.java [hunk 5]
```diff
@@ -135,8 +135,10 @@ public void execute(Event<UIVersionInfo> event) throws Exception {
       String objectId = event.getRequestContext().getRequestParameter(OBJECTID) ;
       uiVersionInfo.curentVersion_  = uiVersionInfo.rootVersion_.findVersionNode(objectId) ;
       UIViewVersion uiViewVersion = uiVersionInfo.getChild(UIViewVersion.class) ;
+      if ( !(uiVersionInfo.curentVersion_.getName().equals("jcr:rootVersion"))) {
       Node frozenNode = uiVersionInfo.curentVersion_.getNode("jcr:frozenNode") ;
       uiViewVersion.setNode(frozenNode) ;
+      }
```
Comment (trongtt) [inline core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/versions/UIVersionInfo.java]: Indentation

File: apps/portlet-explorer/src/main/webapp/groovy/webui/component/explorer/versions/UIVersionInfo.gtmpl [hunk 6]
```diff
@@ -58,19 +51,15 @@
 				println "<a class=\"actionIcon\" onclick=\"" + 
 				         uicomponent.event("RemoveLabel",   node.getPath()) + "\" rel=\"tooltip\" data-placement=\"bottom\" title=\"" + _ctx.appRes("UIVersionInfo.tooltip.removeLable") + "\"><i class=\"uiIconMinus uiIconLightGray\"></i></a>" ;      
 			}         
-			if(!uicomponent.isBaseVersion(node)){
```
Comment (trongtt) [inline apps/portlet-explorer/src/main/webapp/groovy/webui/component/explorer/versions/UIVersionInfo.gtmpl]: why do we remove this condition check ? I understood that the restore link should only appear if the node is not the baseVersion(0)

Comment (AtiaZ) [inline apps/portlet-explorer/src/main/webapp/groovy/webui/component/explorer/versions/UIVersionInfo.gtmpl]: Yes, we are sure that each node is not the baseVersion(0).

**Comment by thomasdelhomenie** [comment]
Merged

**Pull request closed without merging**